### PR TITLE
Doc Updates

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,40 +38,38 @@ If the commit reverts a previous commit, it should begin with `revert:`, followe
...
 
 ### Scope
 
-The scope could be anything specifying place of the commit change. Generally, these should match package names.For example, `http-core`, `common`, `ciscospark`, etc...
+The scope should indicate what is being changed. Generally, these should match package names. For example, `http-core`, `common`, `ciscospark`, etc. Other than package names, `tooling` tends to be the most common.
 
 ### Subject
 
-The subject contains succinct description of the change:
+The subject contains s succinct description of the change:
```
Comment (mwegman) [inline CONTRIBUTING.md]: extra 's'

File: docs/_posts/examples/9999-03-01-storage-layer.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+The storage layer takes care of storing and loading bounded data like user credentials and unbounded data like encryption keys. Note that the storage layer is *not* intended to be your application's database (though may interface with your application's database).
+
+# Usage
+
+Within a spark plugin, the storage layer is available via two properties: `this.boundedStorage` and `this.unboundedStorage`. `boundedStorage` should be used for things with a known size (e.g. user credentials, device registration payloads, etc). `unboundedStorage` should be used for arrays of unknown length (e.g. encryption keys). These properties are bound to the namespace defined by the plugin.
+
+# Adapters
+
+A storage adapter is a proscribed interface to an arbitrary storage backend. Its entry point is `bind()`. Bind accepts one argument, `namespace`, and returns an object containing the methods `del()`, `get()`, and `put()`. These methods are scoped to the `namespace`. For example, the following block would write the value `true` to the key `"proof"` in the `namespace` `"spark-core"`.
```
Comment (mwegman) [inline docs/_posts/examples/9999-03-01-storage-layer.md]: /proscribed/prescribed

Comment (ianwremmel) [inline docs/_posts/examples/9999-03-01-storage-layer.md]: yea, I did that on purpose, but it turns I'm definitely using the wrong word. thanks!

File: README.md [hunk 3]
```diff
@@ -1,43 +1,89 @@
...
+### Serve the package
+This is mostly useful for the the example app(s), but also comes in handy when debugging the credentials plugins automation tests.
+
+```bash
+PACKAGE=PACKAGENAME npm run grunt:package -- serve
+```
+
+### Run unit tests in watch mode
+
+OK, this one's a handfull and requires a global package, but there were too many possible variants to hardcode it any where.
```
Comment (Josh-Dykstra) [inline README.md]: I think handful only has one 'l'

File: README.md [hunk 4]
```diff
@@ -1,43 +1,89 @@
...
+```bash
+UNIT_ONLY=true PACKAGE=PACKAGENAME npm run grunt:package -- test
+```
+
+### Run the tests, but only in nodejs
+Usually faster, and can build on the fly, thus no need to rebuild everything between test runs
 
-Alternatively, you can use use `ciscospark.init()` to set those values at runtime. See Environment Setup (below) and the (ciscospark)[packages/ciscospark/README.md] README for more info on those values.
+```bash
+PACKAGE=PACKAGENAME npm run grunt:package -- express:test test:node
```
Comment (adamweeks) [inline README.md]: The chat-widget doesn't have express tests, should we mention anything about jest tests here?

Comment (ianwremmel) [inline README.md]: We can add detail those tests once the widget merges. express:test won't actually hurt anything, it just stands up a local server to serve some fixtures.

Comment (adamweeks) [inline README.md]: Gotcha, thanks.

**Pull request merged**